shelf/endpoint_decorators.py

```diff
@@ -92,13 +93,16 @@ def log(message, data):
                 container.logger.info("{0} : \n {1}".format(message, data))
 
             if request_data:
-                request_data = json.dumps(
-                    json.loads(
-                        request_data
-                    ),
-                    indent=4,
-                    separators=(',', ': ')
-                )
+                try:
+                    request_data = json.dumps(
+                        json.loads(
+                            request_data
+                        ),
+                        indent=4,
+                        separators=(',', ': ')
+                    )
+                except ValueError as ex:
+                    log("Invalid JSON from request.", ex)
```

Comment:
Ah. I think I just copied how it was logging below that statement. Fixed. 762edd6